api/helpers/game-states/moves/resolve-four/execute.js

```diff
@@ -0,0 +1,65 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Resolve four',
+
+  description: 'Returns new GameState resulting from requested resolve four move',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the requesting player resolves a four card',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to resolve a four
+     * @param { String } requestedMove.cardId1 - First card to be discarded
+     * @param { String } requestedMove.cardId2 - Second card to be discarded
+     * @param { MoveType.RESOLVE_FOUR } requestedMove.moveType - Specifies that this is a Resolve Four move
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'The move being requested. Specifies which player is asking to resolve a four',
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move.',
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState, requestedMove, playedBy }, exits) => {
+    const { cardId1, cardId2 } = requestedMove;
+    let result = _.cloneDeep(currentState);
+
+    const player = playedBy ? result.p1 : result.p0;
+    const discardedCards = [];
+
+    const cardIndex1 = player.hand.findIndex(({ id }) => id === cardId1);
+
+    discardedCards.push(player.hand[cardIndex1]);
+    result.scrap.push(...player.hand.splice(cardIndex1, 1));
```

Comment:
The test failures look to be due to the oneOff (the four itself) not being added to the scrap.
```suggestion
    result.scrap.push(result.oneOff, ...player.hand.splice(cardIndex1, 1));
```

Incidentally, you can see the test runs in cypress cloud at the url linked in the bottom of the github actions output:
![image](https://github.com/user-attachments/assets/1a35ec84-b220-4977-89ad-1f0518a32c44)


In this case this test run is https://cloud.cypress.io/projects/i8bxr8/runs/2077

From there, if you click a test case you can then click 'test replay' to see exactly what happened in CI

![image](https://github.com/user-attachments/assets/214666e6-736a-4db1-acd6-a886199d8c74)


This is super helpful for looking more precisely at why things are failing w/o sifting through terminal logs from the CI output